app.js

```diff
@@ -66,13 +77,24 @@ async function warmupAgendas(agendas) {
 
 async function warmupAgenda(agenda, allowedGroupHeader) {
   const urls = await getAgendaitemsRequestUrls(agenda);
-  for (let url of urls) {
-    await fetch(url, {
-      method: "GET",
-      headers: {
-        "mu-auth-allowed-groups": allowedGroupHeader,
-      },
+  try {
+    await Promise.all(
+      urls.map(async (url) => {
```

Comment:
`async` mag weg

Reply:
moeten we die promises niet awaiten? ik wou per agenda alle calls groeperen, zodat we de parallel calls iets of wat beperken (aantal agendapunten * 2 + 1) calls dus grootste agendas zijn 300 calls.
Ik ben geen krak met promises, als ik `return Promise.all` doe, is dat ook goed want de await van `await warmupAgenda` zou dan volgens mij die groepering respecteren

Reply:
Hier worden wat notaties door elkaar gehaald, wat het wat moeilijk maakt.
Je hebt 2 manieren die equivalent zijn voor promises. Gegeven dat `Promise.all(...)` een promise teruggeeft, kan je ofwel doen:
```javascript
Promise.all(...).then(() => {... }).catch(err => {...})
```
ofwel - en dit is equivalent -
```javascript
try {
  await Promise.all(...)
  // hier wat je in then() doet
} catch (err) {
  // hier wat je in catch() doet
}
```
Het eerste is een oudere notatie dan het tweede. Ik heb persoonlijk een voorkeur voor het tweede want makkelijker leesbaar.

--- 
De inner function heeft geen `await`. Daarom is die `async` overbodig. Die function returnt een promise, aangezien `fetch` een promise teruggeeft. En die wordt ge-await doordat je dit gewrapped hebt in een `await Promise.all(...)`.

Als je een `await` zou zetten bij de fetch, is die `async` wel nodig. Dus dan zou het zijn:
```javascript
async (url) => {
  return await fetch(url, {
      method: "GET",
      headers: {
        "mu-auth-allowed-groups": allowedGroupHeader,
      },
    )
  }
```
In dat geval moet je wel een try-catch block gebruiken ipv. `.then(...).catch(...)`

---

Volgens mij kan je dat hele block reduceren tot
```javascript
    try {
      const promises = urls.map((url) => {
        return fetch(url, {
          method: "GET",
          headers: {
            "mu-auth-allowed-groups": allowedGroupHeader,
          }
        });
      })
      await Promise.all(promises);
    } catch (error) {
      console.warn(`error warming up agenda ${agenda}, not retrying`);
      console.error(error.message);
    }
```